client/src/components/SelectionDialog/SelectionDialog.vue

```diff
@@ -44,21 +44,21 @@ interface Props {
     title?: string;
     searchTitle?: string;
     okButtonText?: string;
-    filterClass?: Filtering<any>;
+    filterClass?: Filtering<unknown>;
 }
 
 const props = withDefaults(defineProps<Props>(), {
     disableOk: false,
     errorMessage: "",
     fileMode: true,
-    fields: () => [],
+    sortKeys: () => defaultSortKeys,
     isBusy: false,
     isEncoded: false,
     items: () => [],
     itemsProvider: undefined,
```

Comment:
I propose to drop the `items` property and manage the items always through the `itemsProvider` function this will eliminate some case handling and will allow proper pagination for all dialogs. It will also ensure proper mapping between the raw items and `SelectionItem`s.

Reply:
I feel like the items provider mechanism could bite us down the road. It's a bit untransparent, and it seems like it could easily lead to bugs like overfetching. I'd rather we try to solve this with standard vue patterns like props+events or a store.

Reply:
You are right, I was still thinking about the old `BTable`. In any case, we just need a common way of "providing" items to the dialog no matter where the data comes from (history, remote file sources, data libraries, etc.), with pagination, sorting and search support (where possible) and to normalize the data to the common `SelectionItem` interface,